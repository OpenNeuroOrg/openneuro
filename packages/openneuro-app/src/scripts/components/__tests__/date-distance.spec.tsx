import React from "react"
import { vi } from "vitest"
import { render } from "@testing-library/react"
import { DateDistance } from "../date-distance"

describe("DataTable component", () => {
  beforeEach(() => {
    vi.useFakeTimers()
    vi.setSystemTime(new Date(2024, 1, 4))
  })
  afterEach(() => {
    vi.useRealTimers()
  })
  it("renders date distance components", () => {
    const date = new Date(2022, 1, 4).toISOString()
    const { asFragment } = render(<DateDistance date={date} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
