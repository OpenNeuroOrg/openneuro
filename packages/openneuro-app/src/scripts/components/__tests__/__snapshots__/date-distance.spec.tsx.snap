// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataTable component > renders date distance components 1`] = `
<DocumentFragment>
  2022-02-04 - about 2 years ago
</DocumentFragment>
`;
